File: examples/quickstart.py
"""Trends API quickstart - Wikipedia. Get a free key at https://trendsapi.ai/#get-key"""
import json, os

import requests

res = requests.post(
    "https://api.trendsapi.ai/api",
    headers={"Authorization": f"Bearer {os.environ['TRENDSAPI_KEY']}"},
    json={
        "mode": "get_growth",
        "source": "wikipedia",
        "keyword": "artificial intelligence",
        "percent_growth": ["3M", "12M"],
    },
    timeout=60,
)
res.raise_for_status()
env = res.json()
print(json.loads(env["body"]) if env.get("statusCode") == 200 else env)

File: examples/quickstart.mjs
// Trends API quickstart - Wikipedia. Get a free key at https://trendsapi.ai/#get-key
const res = await fetch("https://api.trendsapi.ai/api", {
  method: "POST",
  headers: {
    Authorization: `Bearer ${process.env.TRENDSAPI_KEY}`,
    "Content-Type": "application/json",
  },
  body: JSON.stringify({
    mode: "get_growth",
    source: "wikipedia",
    keyword: "artificial intelligence",
    percent_growth: ["3M", "12M"],
  }),
});
if (!res.ok) throw new Error(`HTTP ${res.status}`);
const env = await res.json();
console.log(env.statusCode === 200 ? JSON.parse(env.body) : env);
